xclim/sdba/utils.py

```diff
@@ -922,3 +924,84 @@ def _skipna_correlation(data):
             "allow_rechunk": True,
         },
     ).rename("correlation")
+
+
+def bin_width_estimator(X):
+    """Estimate the bin width of an histogram."""
+    if isinstance(X, list):
+        return np.min([bin_width_estimator(x) for x in X], axis=0)
+
+    if X.ndim == 1:
+        X = X.reshape(-1, 1)
+
+    if X.shape[0] < 1000:
+        # Sturges
+        nh = np.log2(X.shape[0]) + 1.0
+        bin_width = np.zeros(X.shape[1]) + 1.0 / nh
+    else:
+        # Freedman-Diaconis
+        bin_width = (
+            2.0
+            * (np.percentile(X, q=75, axis=0) - np.percentile(X, q=25, axis=0))
+            / np.power(X.shape[0], 1.0 / 3.0)
+        )
+
+    return bin_width
+
+
+def histogram(data, bin_width, bin_origin):
+    """Construct an histogram of the data.
+
+    Returns the position of the center of bins, their corresponding frequency and the bin of every data point.
+    """
+    # Find bin indices of data points
+    idx_bin = np.floor((data - bin_origin) / bin_width)
+
+    # Associate unique values with frequencies
+    idx_grid, mu = np.unique(idx_bin, return_counts=True, axis=0)
+
+    # Normalise frequencies
+    mu = np.divide(mu, sum(mu))
+
+    # Get the unindexed position of the center of bins
+    grid = (idx_grid + 1 / 2) * bin_width + bin_origin
+
+    return grid, mu, idx_bin
+
+
+def optimal_transport(gridX, gridY, muX, muY, numItermax):
+    """Computes the optimal transportation plan between X and Y."""
+    # Compute the distances from every X bin to every Y bin
+    C = distance.cdist(gridX, gridY)
+    C = C**2
```

Comment:
```suggestion
    C = distance.cdist(gridX, gridY, 'sqeuclidean')
```
surely unimportant for overall performance, but why not